src/java/org/apache/cassandra/service/reads/repair/RowIteratorMergeListener.java

```diff
@@ -79,31 +84,32 @@ public RowIteratorMergeListener(DecoratedKey partitionKey, RegularAndStaticColum
         sourceDeletionTime = new DeletionTime[sources.size()];
         markerToRepair = new ClusteringBound[sources.size()];
         this.command = command;
-        this.repairListener = repairListener;
+        this.consistency = consistency;
+        this.readRepair = readRepair;
 
         this.diffListener = new RowDiffListener()
         {
             public void onPrimaryKeyLivenessInfo(int i, Clustering clustering, LivenessInfo merged, LivenessInfo original)
             {
-                if (merged != null && !merged.equals(original))
+                if (merged != null && !merged.equals(original) && !isTransient(i))
```

Comment:
isTransient checks can first since they might be cheaper? I would also pull up the transient value into an array you can reference directly in the listener for performance.